build/oss-fuzz/build.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -eu
+
+# Don't need the stdlib and libc++ gives linker error
+CXXFLAGS="${CXXFLAGS/-stdlib=libc++/}"
+
+# Needed for initialize the submodules, the submodule directories are empty.
+# Some header files (proj_api.h here) do not actually exist in cockroach, it exists in the submodule.
+git submodule update --init --recursive
+
+# Install dependencies for compile_native_go_fuzzer
+go install github.com/AdamKorcz/go-118-fuzz-build@latest
+go get github.com/AdamKorcz/go-118-fuzz-build/testing
+
+bazel clean --expunge
+
+# Generate artifacts for building the binary
+bazel run pkg/gen:code
+bazel run pkg/cmd/generate-cgo:generate-cgo --run_under "cd $SRC/cockroach && "
+
+# File is not being built via bazel and outdated (gives compilation errors in SessionIDEncoding)
+rm pkg/sql/pgwire/fuzz.go
+
+# Rename as go-118-fuzz-build cant read _test.go files (used by sessionID and frontier)
+mv ./pkg/util/span/frontier_test.go pkg/util/span/frontier_test_fuzz.go
+
+# Build the native go fuzz targets
+compile_native_go_fuzzer ./pkg/keys FuzzPrettyPrint fuzzPrettyPrint
+
+compile_native_go_fuzzer ./pkg/sql/sqlliveness/slstorage FuzzSessionIDEncoding fuzzSessionIDEncoding
+
+compile_native_go_fuzzer ./pkg/util/span FuzzBtreeFrontier fuzzBtreeFrontier
+
+# hacky fix since we've already copied these files (used in coverage only)
+sed -i '/cp "\${fuzzer_filename}" "\${OUT}\/rawfuzzers\/\${fuzzer}"/d' /usr/local/bin/compile_native_go_fuzzer > /dev/null 2>&1
+compile_native_go_fuzzer ./pkg/util/span FuzzLLRBFrontier fuzzLLRBFrontier
+
+compile_native_go_fuzzer ./pkg/ccl/pgcryptoccl/pgcryptocipherccl FuzzEncryptDecryptAES fuzzEncryptDecryptAES
+
+compile_native_go_fuzzer ./pkg/ccl/pgcryptoccl/pgcryptocipherccl FuzzNoPaddingEncryptDecryptAES fuzzNoPaddingEncryptDecryptAES
+
+compile_native_go_fuzzer ./pkg/storage FuzzEngineKeysInvariants fuzzEngineKeysInvariants
+
+# Build old fuzz targets which used `gofuzz`
+compile_go_fuzzer /src/cockroach/pkg/util/uuid Fuzz fuzzuuid
+
+# Generate seed corpus
+zip -r $OUT/fuzzPrettyPrint_seed_corpus.zip ./pkg/keys/testdata/fuzz/FuzzPrettyPrint || true
```

Comment:
Currently these directories contain seeds in the native go format, i.e., incompatible with libfuzzer. Would libfuzzer simply ignore them?

Reply:
The only issue is using `f.Add()` to add seed corpus. The seeds in these directories do work and aren't ignored.

Reply:
I guess I'm missing a step which converts the seeds in go format, e.g., [1]?

[1] https://github.com/cockroachdb/cockroach/blob/master/pkg/keys/testdata/fuzz/FuzzPrettyPrint/0993bb3a50de2d9726809ce574e86227a94a7f0c0874449f034c656060f1068e

Reply:
I might be misunderstanding you here but the files in the testdata/fuzz/{FuzzTestName} directory within the package are also treated as seeds and can be added manually in go format or be generated by the fuzzing engine. The example you shared looks like be the generated corpus created by the fuzzer.

Reply:
> The example you shared looks like be the generated corpus created by the fuzzer.

Right, but _not_ `libfuzzer`, whose format is just a single byte array. I'm still confused on how the seeds which aren't in the format `libfuzzer` supports, are being used.

Reply:
Correct, I missed that. Raised the [fix in this PR](https://github.com/cockroachdb/cockroach/pull/126987).